lib/SPIRV/libSPIRV/SPIRVEntry.h

```diff
@@ -857,6 +857,92 @@ template <spv::Op OC> bool isa(SPIRVEntry *E) {
   return E ? E->getOpCode() == OC : false;
 }
 
+template <spv::Op OC>
+class SPIRVContinuedInstINTELBase : public SPIRVEntryNoId<OC> {
+public:
+  template <spv::Op _OC, class T = void>
+  using EnableIfStruct =
+      typename std::enable_if_t<_OC == OpTypeStructContinuedINTEL, T>;
+  template <spv::Op _OC, class T = void>
+  using EnableIfCompositeConst =
+      typename std::enable_if_t<_OC == OpConstantCompositeContinuedINTEL ||
+                                    _OC ==
+                                        OpSpecConstantCompositeContinuedINTEL,
+                                T>;
+  // Complete constructor
+  SPIRVContinuedInstINTELBase(SPIRVModule *M,
+                              const std::vector<SPIRVValue *> TheElements)
+      : SPIRVEntryNoId<OC>(M, TheElements.size() + 1) {
+
+    Elements = SPIRVEntry::getIds(TheElements);
+    validate();
+  }
+
+  SPIRVContinuedInstINTELBase(SPIRVModule *M, unsigned NumOfElements)
+      : SPIRVEntryNoId<OC>(M, NumOfElements + 1) {
+    Elements.resize(NumOfElements);
+    validate();
+  }
+
+  // Incomplete constructor
+  SPIRVContinuedInstINTELBase() : SPIRVEntryNoId<OC>() {}
+
+  template <spv::Op OPC = OC>
+  EnableIfCompositeConst<OPC, std::vector<SPIRVValue *>> getElements() const {
+    return SPIRVEntry::getValues(Elements);
+  }
+
+  template <spv::Op OPC = OC>
+  EnableIfStruct<OPC, SPIRVType *> getMemberType(size_t I) const {
+    return static_cast<SPIRVType *>(SPIRVEntry::getEntry(Elements[I]));
+  }
+
+  SPIRVCapVec getRequiredCapability() const override {
+    return getVec(CapabilityLongConstantCompositeINTEL);
+  }
+
+  llvm::Optional<ExtensionID> getRequiredExtension() const override {
+    return ExtensionID::SPV_INTEL_long_constant_composite;
+  }
+
+  void setElementId(size_t I, SPIRVId Id) { Elements[I] = Id; }
+  SPIRVWord getNumElements() const { return Elements.size(); }
+
+protected:
+  void validate() const override;
+  void setWordCount(SPIRVWord WordCount) override {
+    SPIRVEntry::setWordCount(WordCount);
+    Elements.resize(WordCount - 1);
+  }
+  _SPIRV_DCL_ENCDEC
+
+  std::vector<SPIRVId> Elements;
```

Comment:
About the `clang-tidy` report: some time ago we turned all warnings into errors, our logic was that if particular check is enabled by our config, then we should follow the rules it enforces. However, we didn't know which exact checks are enabled and how it would affect incoming PRs.

As I can see, the approach with protected class members is widely used in the translator, so we either need to refactor the existing code base or disable the clang-tidy check.

You can ignore this error for now until we made a decision (I will submit an issue for that)